astro/src/content/blog/best-practices-for-oauth-in-mobile-apps.mdx

```diff
@@ -0,0 +1,198 @@
+
+*Pros:* 
+- Full control and customization.
+- You own everything
+*Cons:* 
+- Large Dev Ops burden around ID management
+- You own everything
+
+How does [Native Login](/articles/login-authentication-workflows/mobile/native-login-form-to-fusionauth-jwts-refresh-tokens) work? Below you can see how a flow would work with an endpoint at ubyft.com for authentication and one for all of the rideshare data at rides.ubyft.com.
+
+1. Login using your mobile application to your applications authentication endpoint.
+![Diagram of login to site](/img/blogs/best-practices-for-oauth-in-mobile-apps/native-1.png)
+1. Authentication enpdpoint passes back authentication token.
+![Diagram of AT passing to mobile device](/img/blogs/best-practices-for-oauth-in-mobile-apps/native-2.png)
+1. Authentication token is stored on device.
+![Full Screen App](/img/blogs/best-practices-for-oauth-in-mobile-apps/native-3.png)
+1. Authentication token is used to call API endpoints in rides.ubyft.com.
+![Full Screen App](/img/blogs/best-practices-for-oauth-in-mobile-apps/native-4.png)
+1. The rides.ubyft.com application must check that this is a vaid authentication token and then pass back the data in JSON format.
+![Full Screen App](/img/blogs/best-practices-for-oauth-in-mobile-apps/native-5.png)
+
+What do you own during Native Login?
+- All the login flows
+- Securing the password
+- Designing API security
+- Mediating access to third party APIs
+- New auth developments and requirements
+- Siloed user data
+
+### Native Hybrid
+
+Use a third-party identity provider like Firebase but still code the UI
+
+*Pros:* 
+- You don’t own everything
+- You still own UX
+- Leverage third-party ID expertise 
+
+*Cons:* 
+- Vendor lock-in
+- You don't own everything
+
+> A quick not here we have a great article on [Avoiding Authentication System Lock-In](/articles/authentication/avoid-lockin).
+
+1. Login using your mobile application to third-party authentication endpoint.
+![Diagram of login to site](/img/blogs/best-practices-for-oauth-in-mobile-apps/native-hybrid-1.png)
+1. Third-party authentication enpdpoint passes back authentication token.
+![Diagram of AT passing to mobile device](/img/blogs/best-practices-for-oauth-in-mobile-apps/native-hybrid-2.png)
+1. Third-party authentication token is stored on device.
+![Full Screen App](/img/blogs/best-practices-for-oauth-in-mobile-apps/native-hybrid-3.png)
+1. Authentication token is used to call API endpoints in rides.ubyft.com.
+![Full Screen App](/img/blogs/best-practices-for-oauth-in-mobile-apps/native-4.png)
+1. The rides.ubyft.com application must check that this is a vaid authentication token and then pass back the data in JSON format.
+![Full Screen App](/img/blogs/best-practices-for-oauth-in-mobile-apps/native-5.png)
+
+
+### OAuth and OIDC
+
+- Standards-based approach that centralizes ID concerns
+
+*Pros:* 
+- Leverage auth server
```

Comment:
Could add something here like if using a provider, the provider is responsible for login attacks and that sort of thing.